### ISSUE TITLE
Grid format issue

### ISSUE THREAD

**Issue by jousheinfo** [daavid00/pyopmcsp11#2]
Hi,

thanks for sharing the code. When running the code from spe11c, I get .msh, .brep, .geo. However, I am failing to understand how to import any kind of these mesh formats into ResInsight, or how to convert them into ECLIPSE-format. Your help would be very much appreciated. Thank you!

Best regards,
Jose.

**Comment by daavid00** [comment]
Hey,

If you run the spe11c.txt file using pyopmcsp11, then one of the generated folders is called preprocessing, where the input files to run OPM Flow are located, and other folder called output where the files with the results are located (see https://daavid00.github.io/pyopmcsp11/output_folder.html). Then, you could use ResInsight to visualise the results.

**Comment by daavid00** [comment]
That is correct, the current values to generate a coarser grid were set in order to run fast the simulation (ca. 5 minutes, as mentioned in https://daavid00.github.io/pyopmcsp11/examples.html). The grid size in the benchmark description are for reporting the results, then one could also simulate in finer or coarser grids (e.g, if some upscaling methods are used or to have coarser grid size close to the reservoir boundaries). 

The 150 value defines the 150 m of elevation of the curve domain along the y direction, and the 10 value defines the 10 ms of linear elevation as described in the benchmark description. One could set those values to 0 and compare with the current values to see the effect on the grid.

**Comment by jousheinfo** [comment]
Ok, thank you. Another question, the code does not output the faults in the grid as shown in the spe11c task. Is there a way to get the faults as well?

Also, I change the dimensions and N-grids as mentioned before, but the code struggles with convergency, maybe because of the CO2 injecion input. Is there a way to generate only the grid without taking into account the simulation part?

Thank you!

**Comment by daavid00** [comment]
Hei,

If you remove the lines at the bottom in the configuration file where the injection schedule is defined, then it should only generate the grid; otherwise, you can set the injection rates to zero and then the simulation should finish without any convergence issue.

For the current implemented methods to generate the grids, the borders of the faults along the z-direction are not resolved,
only the fault properties are given to the grid cells. It is on the wish list to implement better methods to generate the grids and follow the faults boundaries (we also welcome contributions :) ).

**Comment by jousheinfo** [comment]
Hello,

I have tried these two options, but neither of them generated the grid:

----------------------------------------------

First option:

SWI7    0 SNI7   0 PEN7         0 PENMAX7 3e7 NPOINTS7      2

"""Properties rock"""
"""K [mD], phi [-], thconr [W m-1 K-1]"""
PERM1 0.10132 PORO1 0.10 THCONR1 1.90
PERM2 101.324 PORO2 0.20 THCONR2 1.25
PERM3 202.650 PORO3 0.20 THCONR3 1.25
PERM4 506.625 PORO4 0.20 THCONR4 1.25
PERM5 1013.25 PORO5 0.25 THCONR5 0.92
PERM6 2026.50 PORO6 0.35 THCONR6 0.26
PERM7       0 PORO7    0 THCONR7 2.00

"""Wells radius and position"""
"""radius, x, y, and z position [m] (final positions as well for csp11c)"""
0.15 2700. 1000. 300. 2700. 4000. 300. #Well 1 
0.15 5100. 1000. 700. 5100. 4000. 700. #Well 2

----------------------------------------------

Second option

SWI7    0 SNI7   0 PEN7         0 PENMAX7 3e7 NPOINTS7      2

"""Properties rock"""
"""K [mD], phi [-], thconr [W m-1 K-1]"""
PERM1 0.10132 PORO1 0.10 THCONR1 1.90
PERM2 101.324 PORO2 0.20 THCONR2 1.25
PERM3 202.650 PORO3 0.20 THCONR3 1.25
PERM4 506.625 PORO4 0.20 THCONR4 1.25
PERM5 1013.25 PORO5 0.25 THCONR5 0.92
PERM6 2026.50 PORO6 0.35 THCONR6 0.26
PERM7       0 PORO7    0 THCONR7 2.00

"""Wells radius and position"""
"""radius, x, y, and z position [m] (final positions as well for csp11c)"""
0.15 2700. 1000. 300. 2700. 4000. 300. #Well 1 
0.15 5100. 1000. 700. 5100. 4000. 700. #Well 2

"""Define the injection values 'inj[]'""" 
"""injection time [s], time step size to write results [s], maximum solver time step [s], injected fluid (0 water, 1 co2) (well1), injection rate [kg/s] (well1), temperature [C] (well1), injected fluid (0 water, 1 co2) (well2), ..."""
31536000000 31536000000 315360000 0  0 10 0  0 10
  788400000   788400000  31536000 1  0 10 0  0 10
  788400000   788400000  31536000 1  0 10 1  0 10
29959200000 29959200000 315360000 0  0 10 0  0 10

I would very much appreciate your help!

**Comment by daavid00** [comment]
Hei, I have just updated the main branch and all tests have passed. I have also run locally the spe11c.txt configuration file with  one line in the injection values section (100 100 100 0 0 10 0 0 10) and it runs. You could try to run the configuration file in the tests folder.

**Comment by jousheinfo** [comment]
Hi Sir,

I was trying to run the ECLIPSE DATA file generated when running the OPM file, and it gives errors. I was checking the code, and I think the mistake is in the .mako file template. Keywords from E100 and E300 are mixed. I was trying to modify manually the file, but it still gives me errors (never used ECLIPSE before). One of them is that there are porosity values (=0.5000E+09). Moreover, you also include the keyword OIL in the .mako file. Is this intended? I am attaching the modified DATA file and the file with the errors. Maybe this will be helpful. I remove the OIL to have only WATER and GAS. 

Thank you so much for your time and constant support.

[CSP11C_E300_errors.PRT.txt](https://github.com/daavid00/pyopmcsp11/files/12450993/CSP11C_E300_errors.PRT.txt)
[CSP11C_E300.DATA.txt](https://github.com/daavid00/pyopmcsp11/files/12451006/CSP11C_E300.DATA.txt)

**Comment by daavid00** [comment]
Hei, 

This framework is intended to be run by OPM Flow using the CO2STORE keyword (with OIL as the brine phase, see the OPM Manual, pp 217 https://opm-project.org/?page_id=955). The high porosity values corresponds to the pore volumes multipliers in the boundaries for the spe11b and c cases as described in the benchmark document. It is possible to extend the framework to be compatible with other simulators besides OPM Flow, but unfortunately this is not in our current focus.

**Comment by jousheinfo** [comment]
Hi Sir,

One question, please. The 2D plots (pressure, saturation, mass)  shown after running the text file, as I understood, were plotted for the last time step (after 2050 years) and they are in the XZ plane. But, what I could not understand is the Y-layer. Could you tell me please for which Y-layer these graphs were plotted? Thanks!

Thank you!

**Comment by daavid00** [comment]
Hei,

For the spe11c case this corresponds to the middle layer in the y axis. On October 1st the benchmark description will be final, then we will add the corresponding postprocessing routines to write the results as requested in the benchmark, then one could use the plotting routines provided in the 11thSPE-CSP Github repository for visualisation.

**Comment by jousheinfo** [comment]
Hi sir,

one question please. In the example provided, is constant PT boundary conditions defined in OPM as specified in the description? I could not the proper keyword in the data file.

Moreover, if I want to modify the OPM data file, how can I run it? The code provided runs the text file, which generates the data file. Or should I modify it in the text file only?

Thanks!

Best regards,
Jose.

**Comment by daavid00** [comment]
Hei,

The models used for the PVT tables when the CO2STORE option is activated are explained in the [OPM Manual](https://opm-project.org/?page_id=955) (also [in this presentation](https://opm-project.org/wp-content/uploads/2023/09/co2store_webinar_220923.pdf) you can find more information about the CO2STORE functionality in OPM).

As mentioned in the [documentation](https://daavid00.github.io/pyopmcsp11/output_folder.html), you can modified the generated input decks and after you should be able to run them by executing `flow NAME_OF_DECK.DATA` (see the [tutorials](https://opm-project.org/?page_id=43) in the OPM page).

Since this issue was name as **Grid format issue** and the latests comments are not grid related, it might be better to close it (if there are no more grid related issues) and open new issues if needed (this allows to other users to find easier information if they have similar issues 🙂).
